tests/unit/apex/annotation_check.cpp

```diff
@@ -0,0 +1,226 @@
+//  Copyright (c) 2019 John Biddiscombe
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+// Test that creates a set of tasks using normal priority, but every
+// Nth normal task spawns a set of high priority tasks.
+// The test is intended to be used with a task plotting/profiling
+// tool to verify that high priority tasks run before low ones.
+
+#include <hpx/hpx_init.hpp>
+
+#include <hpx/async.hpp>
+#include <hpx/include/parallel_execution.hpp>
+#include <hpx/include/threads.hpp>
+#include <hpx/lcos/future.hpp>
+#include <hpx/lcos/when_all.hpp>
+#include <hpx/program_options.hpp>
+#include <hpx/runtime/launch_policy.hpp>
+#include <hpx/runtime/threads/executors/pool_executor.hpp>
+#include <hpx/testing.hpp>
+#include <hpx/util/annotated_function.hpp>
+#include "apex_options.hpp"
+
+#include <atomic>
+#include <cstddef>
+#include <iostream>
+#include <string>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+//
+// This test generates a set of tasks with certain names, then checks
+// if the names are present in the screen output from apex.
+// The tasks are spawned using dataflow, or continuations with different
+// launch policies and executors
+// To make scanning the output possible, we prefix the names so that
+// the alphabetical output from apex can be scanned with a regex to
+// check that the expected names are present.
+//
+// See the CMakeLists
+// set_tests_properties( ...  PROPERTIES PASS_REGULAR_EXPRESSION ...)
+
+// --------------------------------------------------------------------------
+// dummy function that just triggers a delay that can be seen in task plots
+void dummy_task(std::size_t n)
+{
+    // no other work can take place on this thread whilst it sleeps
+    bool sleep = true;
+    auto start = std::chrono::steady_clock::now();
+    do
+    {
+        std::this_thread::sleep_for(std::chrono::microseconds(n) / 25);
+        auto now = std::chrono::steady_clock::now();
+        auto elapsed =
+            std::chrono::duration_cast<std::chrono::microseconds>(now - start);
+        sleep = (elapsed < std::chrono::microseconds(n));
+    } while (sleep);
+}
+
+// --------------------------------------------------------------------------
+// string for a policy
+std::string policy_string(const hpx::launch& policy)
```

Comment:
We normally use the east-const notation in our code base (not to start a religious war...). I think we should be consistent.